packages/flutter_test/lib/src/controller.dart

```diff
@@ -699,6 +699,47 @@ abstract class WidgetController {
   ///
   /// Shorthand for `Scrollable.ensureVisible(element(finder))`
   Future<void> ensureVisible(Finder finder) => Scrollable.ensureVisible(element(finder));
+
+  /// Repeatedly scrolls the `scrollable` by `dScroll` until `finder` and
+  /// throws if `finder` is not found for maximum `timeout` times.
+  ///
+  /// Between each scroll, wait for `duration` time for settling.
+  ///
+  /// This is different from [ensureVisible] in that this allows looking for
+  /// `finder` that is not built yet, but requires to specify the scrollable.
+  ///
+  /// If the `scrollable` is infinite and `dScroll`
+  Future<void> scrollUntilVisible(
+    Finder finder,
+    Finder scrollable,
+    double dScroll, {
+      int timeout = 50,
+      Duration duration = const Duration(milliseconds: 50),
+    }
+  ) async {
+    assert(timeout > 0);
+    Offset moveStep;
+    switch(widget<Scrollable>(scrollable).axisDirection) {
+      case AxisDirection.up:
+        moveStep = Offset(0, dScroll);
+        break;
+      case AxisDirection.down:
+        moveStep = Offset(0, -dScroll);
+        break;
+      case AxisDirection.left:
+        moveStep = Offset(dScroll, 0);
+        break;
+      case AxisDirection.right:
+        moveStep = Offset(-dScroll, 0);
+        break;
+    }
+    while(timeout > 0 && finder.evaluate().isEmpty) {
+      await drag(scrollable, moveStep);
+      await pump(duration);
+      timeout -= 1;
+    }
```

Comment:
If we timed out here, shouldn't we throw?

Reply:
I'm imagining throw will happen at `element(finder)`.

Reply:
I added doc on throw.

Reply:
Wouldn't the user want to know that the throw happened here because the timeout was hit? If we let element(finder) throw for us, it doesn't necessarily mean we timed out.

Reply:
Ahh, maybe we should rename timeout to something like `maxScrolls`?

Reply:
Changed `timeout` to `maxScrolls`.

Reply:
If `element` throws, The only way to exit the `while` loop is to trigger timeout. I agree `maxScrolls` should be more reasonable name here.